tests/unit/Core/Framework/DataAbstractionLayer/FieldSerializer/UpdatedByFieldSerializerTest.php

```diff
@@ -0,0 +1,231 @@
+
+        $wrongField = new JsonField('key', 'key');
+
+        $this->expectExceptionObject(DataAbstractionLayerException::invalidSerializerField(UpdatedByField::class, $wrongField));
+
+        $this->fieldSerializer->encode(
+            $wrongField,
+            $existence,
+            $data,
+            $parameters
+        )->current();
+    }
+
+    public function testEncodeWithoutExistingEntity(): void
+    {
+        $data = new KeyValuePair('key', null, false);
+        $existence = $this->createMock(EntityExistence::class);
+        $existence->method('exists')->willReturn(false);
+        $parameters = new WriteParameterBag(
+            $this->createMock(EntityDefinition::class),
+            $this->createWriteContext(Uuid::randomHex()),
+            '/',
+            new WriteCommandQueue(),
+        );
+
+        $result = iterator_to_array($this->fieldSerializer->encode(
+            new UpdatedByField([Context::USER_SCOPE]),
+            $existence,
+            $data,
+            $parameters
+        ));
+
+        static::assertEmpty($result);
+    }
+
+    public function testEncodeWithInvalidScope(): void
+    {
+        $data = new KeyValuePair('key', null, false);
+        $existence = $this->createMock(EntityExistence::class);
+        $existence->method('exists')->willReturn(true);
+
+        $result = 'foo';
+        Context::createDefaultContext()->scope('invalid-scope', function (Context $context) use ($data, $existence, &$result): void {
+            $result = $this->fieldSerializer->encode(
+                new UpdatedByField([Context::USER_SCOPE]),
+                $existence,
+                $data,
+                new WriteParameterBag(
+                    $this->createMock(EntityDefinition::class),
+                    WriteContext::createFromContext($context),
+                    '/',
+                    new WriteCommandQueue(),
+                )
+            )->current();
+        });
+
+        static::assertNull($result);
+    }
+
+    public function testEncodeWithSalesChannelApiSource(): void
+    {
+        $data = new KeyValuePair('key', null, false);
+        $existence = $this->createMock(EntityExistence::class);
+        $existence->method('exists')->willReturn(true);
+        $parameters = new WriteParameterBag(
+            $this->createMock(EntityDefinition::class),
+            $this->createWriteContext(null, Defaults::LIVE_VERSION, false),
+            '/',
+            new WriteCommandQueue(),
+        );
+
+        $result = iterator_to_array($this->fieldSerializer->encode(
+            new UpdatedByField([Context::USER_SCOPE]),
+            $existence,
+            $data,
+            $parameters
+        ));
+
+        static::assertEmpty($result);
+    }
+
+    /**
+     * @deprecated tag:v6.8.0 - remove this test, as the behavior will be removed
+     */
+    public function testEncodeWithNoUserIdDeprecated(): void
+    {
+        Feature::skipTestIfActive('v6.8.0.0', $this);
```

Comment:
can you use `#[DisabledFeatures(['v6.8.0.0'])]` instead?

Reply:
yes, whoops, mixed up unit and integration annotations.